src/Microsoft.AspNet.Server.Kestrel/Http/FrameResponseStream.cs

```diff
@@ -23,13 +23,7 @@ public FrameResponseStream(FrameContext context)
 
         public override bool CanWrite => true;
 
-        public override long Length
-        {
-            get
-            {
-                throw new NotImplementedException();
-            }
-        }
+        public override long Length => Position;
```

Comment:
Both of those could easily be updated. But the idea you suggested of having a request statistics feature should also be considered.